### PR TITLE
[onert-micro] Add training unit tests

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This pr introduces first training unit tests - BostonHousing dataset.

for issue https://github.com/Samsung/ONE/issues/12873
from draft: https://github.com/Samsung/ONE/pull/13107

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

File: onert-micro/onert-micro/src/train/tests/BostonHousingTask.test.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      onert_micro::OMTrainingContext train_context;
+      train_context.batch_size = batch_size;
+      train_context.num_of_train_layers = num_train_layers;
+      train_context.learning_rate = lambda;
+      train_context.loss = loss;
+      train_context.optimizer = train_optim;
+      train_context.beta = beta;
+      train_context.beta_squares = beta_squares;
+      train_context.epsilon = epsilon;
+      train_context.epochs = training_epochs;
```
Comment (chunseoklee) [inline onert-micro/onert-micro/src/train/tests/BostonHousingTask.test.cpp:59]: train_context.num_epoch ?

Comment (BalyshevArtem) [reply onert-micro/onert-micro/src/train/tests/BostonHousingTask.test.cpp:59]: train_context.num_epoch by default before training is zero. Maybe it is bad naming: `epochs` means  the number of epochs, and `num_epoch` means the value of the current epoch during training.

File: onert-micro/onert-micro/src/train/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        OMBackpropExecutionBuilder.cpp
+        losses_functions/MSE.cpp
+        losses_functions/CrossEntropy.cpp
+        metrics/CrossEntropy.cpp
+        metrics/MAE.cpp
+        metrics/MSE.cpp
+        metrics/Accuracy.cpp
+        train_optimizers/SGD.cpp
+        train_optimizers/Adam.cpp
+        tests/BostonHousingTask.test.cpp
```
Comment (chunseoklee) [inline onert-micro/onert-micro/src/train/CMakeLists.txt:23]:
```suggestion
```

**Review by chunseoklee — APPROVED** [review]
other than comment, LGTM

File: onert-micro/onert-micro/include/train/tests/OMTestTrainBase.h [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ONERT_MICRO_TRAIN_TESTS_TEST_TRAIN_BASE_H
+#define ONERT_MICRO_TRAIN_TESTS_TEST_TRAIN_BASE_H
+
+#include <vector>
+
```
Comment (chunseoklee) [inline onert-micro/onert-micro/include/train/tests/OMTestTrainBase.h:21]:
```suggestion
#include <cstddef>
```
Q. which compiler do you use ? In my local, I need this to build.

Comment (BalyshevArtem) [reply onert-micro/onert-micro/include/train/tests/OMTestTrainBase.h:22]: Added this anyway

File: onert-micro/onert-micro/src/train/CMakeLists.txt (hunk 2 above)
Comment (BalyshevArtem) [reply onert-micro/onert-micro/src/train/CMakeLists.txt]: Done

**Comment by BalyshevArtem** [comment]
@chunseoklee, PTAL

**Review by chunseoklee — APPROVED** [review]

**Pull request merged**